valhalla/filesystem.h

```diff
@@ -211,7 +211,8 @@ class directory_entry {
       // entry_->d_reclen =
       // entry_->d_off =
       entry_->d_ino = s.st_ino;
-      strcpy(entry_->d_name, filename.c_str());
+      /* Saving a char* into a std::string */
+      filename = entry_->d_name;
```

Comment:
FYI this is backwards we are trying to put the filename into the directory entries name field